tests/hazmat/primitives/test_aead.py

```diff
@@ -0,0 +1,156 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import, division, print_function
+
+import binascii
+import os
+
+import pytest
+
+from cryptography.exceptions import InvalidTag, UnsupportedAlgorithm
+from cryptography.hazmat.backends.interfaces import CipherBackend
+from cryptography.hazmat.primitives.ciphers.aead import ChaCha20Poly1305
+
+from ...utils import load_nist_vectors, load_vectors_from_file
+
+
+@pytest.mark.supported(
+    only_if=lambda backend: (
+        not backend.chacha20poly1305_supported()
+    ),
+    skip_message="Requires OpenSSL without ChaCha20Poly1305 support"
+)
+@pytest.mark.requires_backend_interface(interface=CipherBackend)
+def test_chacha20poly1305_unsupported_on_older_openssl(backend):
```

Comment:
To use the `supported` mark we need to parametrize with backend even though we never use the arg. Additionally, we have to supply an interface to `requires_backend_interface` even though there is no interface associated with ChaCha20Poly1305 support. So this test file has a lot of pointless arguments that exist only to make the supported mark skip things on the right versions of OpenSSL.